kiva/fonttools/tests/test_font.py

```diff
@@ -193,6 +194,23 @@ def test_font_query_warnings(self):
         self.assertEqual(query.get_weight(), WEIGHT_LIGHT)
         self.assertEqual(query.get_style(), "italic")
 
+    def test_family_queries(self):
+        # regression test for Enable #971
+        # this ensures every font family creates a valid query populated
+        # with query families that work
+        families = [
+            DECORATIVE, DEFAULT, MODERN, ROMAN, SCRIPT, SWISS, TELETYPE
+        ]
+        for family in families:
+            with self.subTest(family=family):
+                font = Font(family=family)
+                query = font._make_font_query()
+                query_family = query.get_family()[0]
+
+                self.assertEqual(query_family, Font.familymap.get(family))
+                self.assertIn(query_family, font_family_aliases)
+                self.assertIn(query_family, preferred_fonts)
```

Comment:
There are items in `font_family_aliases` which are not keys in `preferred_fonts`. Perhaps you should only check against `preferred_fonts`?

Reply:
We want to check it is in both, because of this line here: https://github.com/enthought/enable/blob/4d51db21a3cae2e316afa3c7329e35eba26ea589/kiva/fonttools/_score.py#L38

If it is not on `font_family_aliases` then `preferred_fonts` will never be accessed.

Reply:
I'm saying that `query_family in font_family_aliases` can be `True` at the same time that `query_family in preferred_fonts` is `False`.

The more complete snippet from `score_family`...
https://github.com/enthought/enable/blob/4d51db21a3cae2e316afa3c7329e35eba26ea589/kiva/fonttools/_score.py#L38-L40

... is not really being tested here because converting from the integer Kiva family constants via `Font.familymap` will never result in the values `"sans"`, `"sans serif"`, or `"modern"`.

**Is the point of this test to keep `Font.familymap` values in sync with `font_family_aliases` and `preferred_fonts`?**

Reply:
The entries in `familymap` need to be in both, yes (and that's what I care about for this PR).

`font_family_aliases` (and presumably `preferred_fonts`) can have extra values which come from the `face_name` (this is what derailed #977).  Eg. if someone names a font "sans" (ie. `Font("sans", 12)`) it will _also_ go through the above codepath.  That likely needs some re-engineering at some point, but it is outside the scope of this PR)